src/templates/dso_api/dynamic_api/api.html

```diff
@@ -63,8 +67,458 @@
+  }
+
+  function getData(url) {
+    return new Promise(function(callback, err) {
+      let http = new XMLHttpRequest();
+      http.open("GET", url, true);
+      http.setRequestHeader("Accept", "application/hal+json");
+      http.send();
+      http.onreadystatechange = function () {
+        if (this.readyState == 2){
+            parseHeaders(this);
+        }
+        if (this.readyState == 4){
+          let result = JSON.parse(this.responseText);
+          try {   
+            callback(result);
+          } catch (error) {
+              err("Error: JSON corrupt");
+          }
+        }
+      }; 
+    }); 
+  }
+
+  function parseHeaders(response) {
+    // Set header fields
+    let headersElement = document.getElementById("response-headers");
+    headersElement.innerHTML = "<b>HTTP " + response.status + " " + response.statusText + "</b>\n";
+    
+    for(let header of HEADERS) {
+      headerValue = response.getResponseHeader(header);
+      if (headerValue){
+        let headerline =  "<b>" + header + ':</b> <span class="lit">' + response.getResponseHeader(header) + "</span>\n";
+        headersElement.innerHTML += headerline;
+      }
+    }
+    
+    // Remove page links if no pagination (for error pages)
+    let page = response.getResponseHeader("X-Pagination-Page");
+    if (!page) {
+      document.getElementById("page-container").classList.remove("active");                                   
+    } else if (!isPaginated) {
+      // If paginated but pagination has not yet been set (for page 1)
+      setPageLinks(page);
+    }
+  }
+
+  function exceptionHandler(e) {
+      console.log("Kan data niet ophalen.");
+  }
+
+  function parseData(data) {
+    let contentElement = document.getElementById("response-content");
+
+    // We need to mark coordinates fields so we can make them collapsible.
+    let markCoordinates = (key, value) => {
+        if (key == "coordinates"){
+            return "__coordinatestring__" + JSON.stringify(value, null, 4);
+        }
+        return value;
+    }
+    
+    // Activate next link
+    if(data["_links"] && data["_links"]['next']) {
+      document.getElementById("page-links").getElementsByTagName("li").item(6).classList.remove("disabled");
+    }
+
+    // Create highlighted html string and set content
+    jsonString = JSON.stringify(data, markCoordinates,4);
+    contentElement.innerHTML = syntaxHighlight(jsonString);
+  }
+
+  function syntaxHighlight(jsonString) {
+      jsonString = jsonString.replace(/&/g, '&amp;').replace(/</g, '&lt;').replace(/>/g, '&gt;');
+      return jsonString.replace(/("(\\u[a-zA-Z0-9]{4}|\\[^u]|[^\\"])*"(\s*:)?|\b(true|false|null)\b|-?\d+(?:\.\d*)?(?:[eE][+\-]?\d+)?)/g, function (match) {
+          let cls = 'number';
+          if (/^"https?:\/\//.test(match)) {
+              // Create a link for strings containing an url.
+              return '<a href=' + match + ' rel="nofollow"><span class="link">' + match +'</span></a>'
```

Comment:
link zonder quotes?

Reply:
de string bevat zelf nog quotes.

Reply:
Als iemand nu de pagina aanroept met `?...&foo=<img onerror="alert(1)">` faciliteren wij een XSS op dataportaal omdat we query parameters ook weer laten terugkomen in de response.

Reply:
Oja in die _links. Zoiets dan?  `<a href="${match.substring(1,match.length-1).replace(/\%3C/g, '').replace(/\%3E/g,'')}" rel="nofollow"><span class="link">${match}</span></a>`